lib/pages/events_page.dart

```diff
@@ -76,16 +111,7 @@ class _EventsPageState extends State<EventsPage> {
         child: IndexedStack(
```

Comment:
We can add something like a stack view here as a child of the root widget, so we can put the empty state overlay on top of it inside a visibility widget, so we can flip its visibility when;

- it's not loading
- when active tab index has 0 items

and the widget three would be something like;

- EventsPage
  - StackView
  - Visibility
    - EmptyState
  - IndexedStack
    - ...

Reply:
Not necessarily Visibility tho, i couldn't think of anything better from top of my head. Please feel free to experiment whit anything you like.

Reply:
Hi, I tried to use `AnimatedCrossFade` and set duration to `0` for now. It's nice to have the fading effect but for some reason flutter doesn't render it properly.

Reply:
Yea i assume overlay has to always exist on the widget tree, then we won't need the shrinkwrap stuff

Reply:
Should we experiment with this approach if you like it?
https://github.com/eralpkaraduman/event_dot_pizza/pull/27#discussion_r356049505

Reply:
Regarding second part of https://github.com/eralpkaraduman/event_dot_pizza/pull/27#issuecomment-564907126

This looks like a job for `AbsorbPointer`
https://api.flutter.dev/flutter/widgets/AbsorbPointer-class.html

So we could;
```
EventsPage(
  StackView(
    AbsorbPointer(absorbing: _noEventsFound,
      Visibility( visible: _noEventsFound,
        EmptyState()
      )
   IndexedStack( ... )
   )
)
```

I don't know if AbsorbPointer is necessary when we use Visibility tho, maybe it absorbs touches auto.?

Reply:
btw, I couldn't find `StackView` class from flutter, I only see `Stack` class.  https://api.flutter.dev/flutter/widgets/Stack-class.html :-?

Reply:
Yes! Sorry for the confusion, i meant exactly that one.

Reply:
Seems like `Stack` approach solves two problem at once, now I can pull to refresh the list also.

Reply:
Awesome 👍